spring-cloud-skipper-server-core/src/main/java/org/springframework/cloud/skipper/server/config/security/OAuthSecurityConfiguration.java

```diff
@@ -0,0 +1,204 @@
+import org.springframework.context.annotation.Conditional;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.context.event.EventListener;
+import org.springframework.security.authentication.AuthenticationManager;
+import org.springframework.security.authentication.AuthenticationProvider;
+import org.springframework.security.authentication.ProviderManager;
+import org.springframework.security.config.annotation.web.builders.HttpSecurity;
+import org.springframework.security.config.annotation.web.configuration.EnableWebSecurity;
+import org.springframework.security.config.annotation.web.configuration.WebSecurityConfigurerAdapter;
+import org.springframework.security.config.annotation.web.configurers.ExpressionUrlAuthorizationConfigurer;
+import org.springframework.security.oauth2.client.OAuth2ClientContext;
+import org.springframework.security.oauth2.client.OAuth2RestTemplate;
+import org.springframework.security.oauth2.client.filter.OAuth2AuthenticationFailureEvent;
+import org.springframework.security.oauth2.client.filter.OAuth2ClientAuthenticationProcessingFilter;
+import org.springframework.security.oauth2.client.token.grant.code.AuthorizationCodeResourceDetails;
+import org.springframework.security.oauth2.config.annotation.web.configuration.EnableOAuth2Client;
+import org.springframework.security.oauth2.provider.authentication.OAuth2AuthenticationManager;
+import org.springframework.security.oauth2.provider.authentication.OAuth2AuthenticationProcessingFilter;
+import org.springframework.security.web.authentication.LoginUrlAuthenticationEntryPoint;
+import org.springframework.security.web.authentication.www.BasicAuthenticationEntryPoint;
+import org.springframework.security.web.authentication.www.BasicAuthenticationFilter;
+import org.springframework.security.web.util.matcher.AntPathRequestMatcher;
+import org.springframework.security.web.util.matcher.AnyRequestMatcher;
+
+/**
+ * Setup Spring Security OAuth for the Rest Endpoints of Spring Cloud Data Flow.
+ *
+ * @author Gunnar Hillert
+ * @author Ilayaperumal Gopinathan
+ */
+@EnableOAuth2Client
+@EnableWebSecurity
+@Configuration
+@Conditional(OnSecurityEnabledAndOAuth2Enabled.class)
+public class OAuthSecurityConfiguration extends WebSecurityConfigurerAdapter {
+
+	private static final org.slf4j.Logger logger = LoggerFactory.getLogger(OAuthSecurityConfiguration.class);
+
+	@Autowired
+	private SecurityStateBean securityStateBean;
+
+	@Autowired
+	private SecurityProperties securityProperties;
+
+	@Autowired
+	private OAuth2ClientContext oauth2ClientContext;
+
+	@Autowired
+	private AuthorizationCodeResourceDetails authorizationCodeResourceDetails;
+
+	@Autowired
+	private ResourceServerProperties resourceServerProperties;
+
+	@Autowired
+	private ApplicationEventPublisher applicationEventPublisher;
+
+	@Autowired
+	private AuthorizationProperties authorizationProperties;
+
+	@Override
+	protected void configure(HttpSecurity http) throws Exception {
```

Comment:
Based on the changes on this class, I believe we don't don't need to create the entire `OAuthSecurityConfiguration` class again here. Here are couple of possible options:

1) Extend the existing `OAuthSecurityConfiguration` and customize the `configure` method.
2) Modify `OAuthSecurityConfiguration` to inject `RequestMatcher`s like `BrowserDetectingContentNegotiationStrategy` programmatically and use that `OAuthSecurityConfiguration` here.

While we can go with option 1) for now, I can work on using Option 2) to make changes in the common security config.

In both ways, we can avoid re-creating this configuration class whole again.